iroh-net/src/magic_endpoint.rs

```diff
@@ -649,25 +678,130 @@ impl MagicEndpoint {
+    pub fn into_0rtt(self) -> Result<(quinn::Connection, quinn::ZeroRttAccepted), Self> {
+        match self.inner.into_0rtt() {
+            Ok((conn, zrtt_accepted)) => {
+                // If we can't notify the rtt-actor that's not great but not critical.
+                let Ok(peer_id) = get_remote_node_id(&conn) else {
+                    warn!(?conn, "failed to get remote node id");
+                    return Ok((conn, zrtt_accepted));
+                };
+                let Ok(conn_type_changes) = self.magic_ep.conn_type_stream(&peer_id) else {
+                    warn!(?conn, "failed to create conn_type_stream");
+                    return Ok((conn, zrtt_accepted));
+                };
+                let rtt_msg = RttMessage::NewConnection {
+                    connection: conn.weak_handle(),
+                    conn_type_changes,
+                    node_id: peer_id,
+                };
+                if let Err(err) = self.magic_ep.rtt_actor.msg_tx.try_send(rtt_msg) {
+                    warn!(?conn, "rtt-actor not reachable: {err:#}");
+                }
+                Ok((conn, zrtt_accepted))
+            }
+            Err(inner) => Err(Self {
+                inner,
+                magic_ep: self.magic_ep,
+            }),
+        }
+    }
+
+    /// Parameters negotiated during the handshake
+    pub async fn handshake_data(&mut self) -> Result<Box<dyn Any>, quinn::ConnectionError> {
+        self.inner.handshake_data().await
+    }
+
+    /// The local IP address which was used when the peer established the connection.
+    pub fn local_ip(&self) -> Option<IpAddr> {
+        self.inner.local_ip()
+    }
+
+    /// The peer's UDP address.
+    pub fn remote_address(&self) -> SocketAddr {
+        self.inner.remote_address()
+    }
+
+    /// Extracts the ALPN protocol from the peer's handshake data.
+    // Note, we could totally provide this method to be on a Connection as well.  But we'd
+    // need to wrap Connection too.
+    pub async fn alpn(&mut self) -> Result<String> {
+        let data = self.handshake_data().await?;
+        match data.downcast::<quinn::crypto::rustls::HandshakeData>() {
+            Ok(data) => match data.protocol {
+                Some(protocol) => std::string::String::from_utf8(protocol).map_err(Into::into),
+                None => bail!("no ALPN protocol available"),
+            },
+            Err(_) => bail!("unknown handshake type"),
+        }
+    }
+}
+
+impl Future for Connecting {
+    type Output = Result<quinn::Connection, quinn::ConnectionError>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut std::task::Context<'_>) -> Poll<Self::Output> {
+        let this = self.project();
+        match this.inner.poll(cx) {
+            Poll::Pending => Poll::Pending,
+            Poll::Ready(Err(err)) => Poll::Ready(Err(err)),
+            Poll::Ready(Ok(conn)) => {
+                // If we can't notify the rtt-actor that's not great but not critical.
```

Comment:
wrong comment?

Reply:
This comment applies to all the `let Ok(...) = ...` statements that follow.  But clearly it's confusing.  Any suggestion on how to explain this better?

Reply:
maybe a helper function that these calls? 

```rust
fn try_send_rtt_msg(conn, magic_ep) {
  let Ok(peer_id) = get_remote_node_id(&conn) else {
    warn!(?conn, "failed to get remote node id");
    return;
  };
  let Ok(conn_type_changes) = this.magic_ep.conn_type_stream(&peer_id) else {
    warn!(?conn, "failed to create conn_type_stream");
    return;
  };
  let rtt_msg = RttMessage::NewConnection {
    connection: conn.weak_handle(),
   conn_type_changes,
   node_id: peer_id,
  };
  if let Err(err) = this.magic_ep.rtt_actor.msg_tx.try_send(rtt_msg) {
    warn!(?conn, "rtt-actor not reachable: {err:#}");
  }
  Ok(())
}
```